src/main/cpp/Robot.cpp

```diff
@@ -82,6 +95,65 @@ void Robot::ExpectAutonomousEndConds() {
     }
 }
 
+void Robot::ClimbingSequenceSM() {
+    switch (m_state) {
+        case ClimbingStates::kSecondRung: {
+            solenoidTimer.Start();
+            if (solenoidTimer.HasElapsed(1_s)) {
+                climber.TelescopingOut();
+                solenoidTimer.Stop();
+                solenoidTimer.Reset();
+            } else if ((!climber.BarSensor()) && (!climber.UppperSensor())) {
+                climber.TelescopingExtention(0.65);
+            }
+
+            if ((climber.BarSensor()) && (!climber.LowerSensor())) {
+                climber.TelescopingExtention(-0.65);
+            }
+
+            if (climber.LowerSensor()) {
+                climber.TelescopingExtention(0.00);
+                climber.TelescopingIn();
+                m_state = ClimbingStates::kThridRung;
+            }
+            break;
+        }
+        case ClimbingStates::kThridRung: {
+            solenoidTimer.Start();
+            if (solenoidTimer.HasElapsed(1_s)) {
+                climber.TelescopingOut();
+                solenoidTimer.Stop();
+                solenoidTimer.Reset();
+            } else if ((!climber.BarSensor()) && (!climber.UppperSensor())) {
+                climber.TelescopingExtention(0.65);
+            }
+
+            if ((climber.BarSensor()) && (!climber.LowerSensor())) {
+                climber.TelescopingExtention(-0.65);
+            }
+
+            if (climber.LowerSensor()) {
+                climber.TelescopingExtention(0.00);
+                climber.TelescopingIn();
+                m_state = ClimbingStates::kTraversalRung;
+            }
```

Comment:
Consider making this a function as it's repeated in two states and writing it a bunch is cumbersome.